### PR TITLE
ARCHETYPE-536 Update Groovy Dependency and Include Ivy Dependency

### PR DESCRIPTION
**Pull request by stefanseifert**
see https://issues.apache.org/jira/browse/ARCHETYPE-536

### CODE REVIEW

File: archetype-common/pom.xml [hunk 1]
```diff
@@ -174,7 +174,20 @@
     <dependency>
       <groupId>org.codehaus.groovy</groupId>
       <artifactId>groovy</artifactId>
-      <version>1.8.3</version>
+      <version>2.4.9</version>
```
Comment (hboutemy) [inline archetype-common/pom.xml]: please upgrade to latest 2.4.x

File: archetype-common/pom.xml [hunk 2]
```diff
@@ -174,7 +174,20 @@
...
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.ivy</groupId>
+      <artifactId>ivy</artifactId>
+      <version>2.4.0</version>
+      <scope>compile</scope>
```
Comment (hboutemy) [inline archetype-common/pom.xml]: scope should be runtime

File: maven-archetype-plugin/src/it/projects/ARCHETYPE-536_groovy-grape/archetype/pom.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  </parent>
+  <artifactId>archetype536-archetype</artifactId>
+  <packaging>maven-archetype</packaging>
+  
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-archetype-plugin</artifactId>
+        <extensions>true</extensions>
```
Comment (hboutemy) [inline maven-archetype-plugin/src/it/projects/ARCHETYPE-536_groovy-grape/archetype/pom.xml]: why extension=true? not a good idea on a plugin with so many dependencies

**Comment by stefanseifert** [comment]
thanks for the review!
i fixed the three issue you mentioned and merged the PR with the master branch.

**Comment by hboutemy** [comment]
great
please rebase and squash commits

**Comment by stefanseifert** [comment]
done!

**Comment by hboutemy** [comment]
merged, thank you

**Comment by jira-importer** [comment]
Resolve #766

**Comment by jira-importer** [comment]
Resolve #766

**Comment by jira-importer** [comment]
Resolve #766

**Pull request merged**